src/main/java/de/tum/in/www1/artemis/web/rest/StudentQuestionAnswerResource.java

```diff
@@ -118,35 +139,47 @@ public StudentQuestionAnswerResource(StudentQuestionAnswerRepository studentQues
     }
 
     /**
-     * GET /question-answers/:id : get the "id" questionAnswer.
+     * GET /courses/{courseId}/question-answers/:id : get the "id" questionAnswer.
      *
+     * @param courseId the id of the course the answer belongs to
      * @param id the id of the questionAnswer to retrieve
      * @return the ResponseEntity with status 200 (OK) and with body the questionAnswer, or with status 404 (Not Found)
      */
-    @GetMapping("/student-question-answers/{id}")
+    @GetMapping("courses/{courseId}/student-question-answers/{id}")
     @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
-    // TODO: there are no security checks here. The API endpoint should at least include the course id
-    public ResponseEntity<StudentQuestionAnswer> getStudentQuestionAnswer(@PathVariable Long id) {
+    public ResponseEntity<StudentQuestionAnswer> getStudentQuestionAnswer(@PathVariable Long courseId, @PathVariable Long id) {
         log.debug("REST request to get StudentQuestionAnswer : {}", id);
+        User user = this.userService.getUserWithGroupsAndAuthorities();
+        Optional<Course> optionalCourse = courseRepository.findById(courseId);
+        if (optionalCourse.isEmpty()) {
+            return ResponseEntity.notFound().build();
+        }
+        if (!this.authorizationCheckService.isAtLeastStudentInCourse(optionalCourse.get(), user)) {
+            return forbidden();
+        }
         Optional<StudentQuestionAnswer> questionAnswer = studentQuestionAnswerRepository.findById(id);
         return ResponseUtil.wrapOrNotFound(questionAnswer);
     }
 
     /**
-     * DELETE /question-answers/:id : delete the "id" questionAnswer.
+     * DELETE /courses/{courseId}/question-answers/:id : delete the "id" questionAnswer.
      *
+     * @param courseId the id of the course the answer belongs to
      * @param id the id of the questionAnswer to delete
      * @return the ResponseEntity with status 200 (OK)
      */
-    @DeleteMapping("/student-question-answers/{id}")
-    // TODO: there are no security checks here. The API endpoint should at least include the course id
+    @DeleteMapping("courses/{courseId}/student-question-answers/{id}")
     @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
-    public ResponseEntity<Void> deleteStudentQuestionAnswer(@PathVariable Long id) {
+    public ResponseEntity<Void> deleteStudentQuestionAnswer(@PathVariable Long courseId, @PathVariable Long id) {
         User user = userService.getUserWithGroupsAndAuthorities();
         Optional<StudentQuestionAnswer> optionalStudentQuestionAnswer = studentQuestionAnswerRepository.findById(id);
         if (optionalStudentQuestionAnswer.isEmpty()) {
             return ResponseEntity.notFound().build();
         }
+        Optional<Course> optionalCourse = courseRepository.findById(courseId);
+        if (optionalCourse.isEmpty()) {
+            return ResponseEntity.notFound().build();
+        }
```

Comment:
Similar to above, this does not check if the requested question is in the same course as `courseId`.